sycamore/docset.py

```diff
@@ -529,6 +577,26 @@ def map_batch(
         )
         return DocSet(self.context, map_batch)
 
+    def map_elements(self, f: Callable[[Element], Element], **resource_args) -> "DocSet":
+        """
+        Applies the given mapping function to each element in the each Document in this DocsSet.
+
+        Args:
+            f: A Callable that takes an Element and returns an Element. Elements for which
+                f evaluates to None are dropped.
+        """
+
+        def process_doc(doc: Document) -> Document:
+            new_elements = []
+            for e in doc.elements:
```

Comment:
Given the current data model I don't think we can parallelize/distribute this to a map call over elements right? Docs can have a large amount of elements and ops can get expensive

Reply:
Yeah, this doesn't solve the problem of a huge number of elements in a single document. At the moment I don't think we have a better mechanism for that than exploding and processing the elements as documents.